powerdns/models/powerdns.py

```diff
@@ -591,17 +595,21 @@ def create_ptr(self):
         else:
             return
 
-        self.delete_ptr()
-        Record.objects.create(
-            type='PTR',
-            domain=domain,
-            name='.'.join([number, domain_name]),
-            content=self.name,
-            depends_on=self,
-            owner=self.owner,
-            ttl=self.ttl,
-            disabled=self.disabled,
-        )
+        # ensure 1 PTR (at the most) exist for the same IP
+        if not Record.objects.filter(
```

Comment:
couldn't we use here https://docs.djangoproject.com/en/1.9/ref/models/querysets/#update-or-create ?